FV3/atmos_cubed_sphere/tools/coarse_grained_diagnostics.F90

```diff
@@ -43,8 +44,8 @@ module coarse_grained_diagnostics_mod
   public :: fv_coarse_diag_init, fv_coarse_diag
 
   integer :: tile_count = 1  ! Following fv_diagnostics.F90
-  integer :: DIAG_SIZE = 512
-  type(coarse_diag_type), dimension(512) :: coarse_diagnostics
+  integer :: DIAG_SIZE = 1024
+  type(coarse_diag_type), dimension(1024) :: coarse_diagnostics
```

Comment:
This was a bug that had been lurking in fv3gfs-fortran for a while now and it was something I fixed in SHiELD way back in January of 2021 (prior to the PIRE simulation).  It only surfaces when running with the TKE-EDMF scheme turned on and with coarse diagnostics active, which is why we have not encountered it up to this point (my C384 runs were done without the TKE-EDMF scheme).

Since the code dynamically adds diagnostics for any tracers, the more tracers you have the more diagnostics you have.  Since we add diagnostics for each tracer interpolated to 31 different pressure levels, this can cause the number of diagnostics to accumulate fairly quickly.  When running with the TKE-EDMF scheme, which uses an extra tracer, there are actually 522 possible coarse diagnostics, which is more than the 512 allocated here.  

There shouldn't be any significant performance implications from increasing the allocation since no arrays are stored in this data structure (just some metadata and some pointers), so to address this issue I went ahead and doubled the size of this derived-type array to 1024.

Reply:
I think I will split this off into another PR first for clarity, since it is somewhat unrelated to the rest of the changes.

Reply:
I split this off into #337.